src/queues/distributed_queue.js

```diff
@@ -176,6 +176,22 @@ class Queue {
         })
     }
 
+    /**
+     * Wrapper around 'publishTo'. Will send delayed message.
+     * If publishing to an exchange that was not declared with 'delayedMessages' it will do nothing
+     * and the message will et delivered immediately.
+     * @param {string} routingKey the name of the queue or topic to publish to
+     * @param {string} message the message to publish
+     * @param {*} [options] the options to attach to the published message
+     * @param {number} delay Milliseconds to delay the message
+     * @returns {PromiseLike<any>}
+     */
+    async publishWithDelayTo(routingKey, message, options, delay) {
```

Comment:
```suggestion
    async publishWithDelayTo(routingKey, message, delay, options) {
```

Since `delay` is not optional and `options` is optional, delay should come before, otherwise you force me to pass null arguments

Reply:
IMO, since the `publishTo` method already provides named options, adding a `publishWithDelayTo` is unnecessary. This really means that if in my development I decide to change from a regular channel to a delayed channel, that I need to refactor my publishers to use a different method. Why not just add the delayed option to the `publshTo`? Then you could (possibly later) add a `defaultDelay` configuration, so that in most cases I won't even need to worry about passing this parameter. Seems more flexible to me, and you could still provide descriptive error if you want.